src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -167,8 +168,8 @@ def activate_osrandom_engine(self):
                 # Set the engine as the default RAND provider.
                 res = self._lib.ENGINE_set_default_RAND(e)
                 self.openssl_assert(res == 1)
-            # Reset the RNG to use the new engine.
-            self._lib.RAND_cleanup()
+            # Reset the RNG to use the engine
+            res = self._lib.RAND_set_rand_method(self._ffi.NULL)
```

Comment:
I do not understand this at all.

And unchecked `res`

Reply:
Take a look at the source code and it'll sort of make sense. Or, if not, look at the implementation of `RAND_cleanup` in 1.0.x and you'll see this was already called.

I'll add the assert momentarily.